docs/how-to/hip_porting_guide.rst

```diff
@@ -0,0 +1,649 @@
+hipified code to ``stdout``.
+
+``hipconvertinplace.sh``or  ``hipconvertinplace-perl.sh`` operate on whole
+directories.
+
+Library Equivalents
+================================================================================
+
+ROCm provides libraries to ease porting of code relying on CUDA libraries.
+Most CUDA libraries have a corresponding HIP library.
+
+There are two flavours of libraries provided by ROCm, ones prefixed with ``hip``
+and ones prefixed with ``roc``. While both are written using HIP, in general
+only the ``hip``-libraries are portable. The libraries with the ``roc``-prefix
+might also run on CUDA-capable GPUs, however they have been optimized for AMD
+GPUs and might use assembly code or a different API, to achieve the best
+performance.
+
+.. note::
+
+  If the application is only required to run on AMD GPUs, it is recommended to
+  use the ``roc``-libraries.
+
+In the case where a library provides a ``roc``- and a ``hip``- version, the
+``hip`` version is a marshalling library, which is just a thin layer that is
+redirecting the function calls to either the ``roc``-library or the
+corresponding CUDA library, depending on the platform, to provide compatibility.
+
+.. list-table::
+  :header-rows: 1
+
+  *
+   - CUDA Library
+   - ``hip`` Library
+   - ``roc`` Library
+   - Comment
+  *
+   - cuBLAS
+   - `hipBLAS <https://github.com/ROCm/hipBLAS>`_
+   - `rocBLAS <https://github.com/ROCm/rocBLAS>`_
+   - Basic Linear Algebra Subroutines
+  *
+   - cuBLASLt
+   - `hipBLASLt <https://github.com/ROCm/hipBLASLt>`_
+   -
+   - Linear Algebra Subroutines, lightweight and new flexible API
+  *
+   - cuFFT
+   - `hipFFT <https://github.com/ROCm/hipFFT>`_
+   - `rocFFT <https://github.com/ROCm/rocfft>`_
+   - Fast Fourier Transfer Library
+  *
+   - cuSPARSE
+   - `hipSPARSE <https://github.com/ROCm/hipSPARSE>`_
+   - `rocSPARSE <https://github.com/ROCm/rocSPARSE>`_
+   - Sparse BLAS + SPMV
+  *
+   - cuSOLVER
+   - `hipSOLVER <https://github.com/ROCm/hipsolver>`_
+   - `rocSOLVER <https://github.com/ROCm/rocsolver>`_
+   - Lapack library
+  *
+   - AmgX
+   -
+   - `rocALUTION <https://github.com/ROCm/rocalution>`_
+   - Sparse iterative solvers and preconditioners with algebraic multigrid
+  *
+   - Thrust
+   -
+   - `rocThrust <https://github.com/ROCm/rocThrust>`_
+   - C++ parallel algorithms library
+  *
+   - CUB
+   - `hipCUB <https://github.com/ROCm/hipcub>`_
+   - `rocPRIM <https://github.com/ROCm/rocPRIM>`_
+   - Low Level Optimized Parallel Primitives
+  *
+   - cuDNN
+   -
+   - `MIOpen <https://github.com/ROCm/MIOpen>`_
+   - Deep learning Solver Library
+  *
+   - cuRAND
+   - `hipRAND <https://github.com/ROCm/hiprand>`_
+   - `rocRAND <https://github.com/ROCm/rocrand>`_
+   - Random Number Generator Library
+  *
+   - NCCL
+   -
+   - `RCCL <https://github.com/ROCm/rccl>`_
+   - Communications Primitives Library based on the MPI equivalents
+     RCCL is a drop-in replacement for NCCL
+
```

Comment:
EIGEN shouldn't be mentioned any more here?

Reply:
I can't see the mention of EIGEN. Where are we mentioning?

Reply:
Now I understand. Sent message directly.